pyomo/gdp/plugins/bigm.py

```diff
@@ -706,3 +788,131 @@ def _estimate_M(self, expr, name):
                 M = (expr_lb, expr_ub)
 
         return tuple(M)
+
+    # These are all functions to retrieve transformed components from original
+    # ones and vice versa.
+    def get_src_disjunct(self, transBlock):
+        """Return the Disjunct object whose transformed components are on
+        transBlock.
+
+        Parameters
+        ----------
+        transBlock: _BlockData which is in the relaxedDisjuncts IndexedBlock
+                    on a transformation block.
+        """
+        if not hasattr(transBlock, '_srcDisjunct') or \
+           not type(transBlock._srcDisjunct) is weakref_ref:
+            raise GDP_Error("Block %s doesn't appear to be a transformation "
+                            "block for a disjunct. No source disjunct found." 
+                            % transBlock.name)
+        return transBlock._srcDisjunct()
+
+    def get_src_constraint(self, transformedConstraint):
+        """Return the original Constraint whose transformed counterpart is
+        transformedConstraint
+
+        Parameters
+        ----------
+        transformedConstraint: Constraint, which must be a component on one of 
+        the BlockDatas in the relaxedDisjuncts Block of 
+        a transformation block
+        """
+        transBlock = transformedConstraint.parent_block()
+        # This should be our block, so if it's not, the user messed up and gave
+        # us the wrong thing. If they happen to also have a _constraintMap then
+        # the world is really against us.
+        if not hasattr(transBlock, "_constraintMap"):
+            raise GDP_Error("Constraint %s is not a transformed constraint" 
+                            % transformedConstraint.name)
+        return transBlock._constraintMap['srcConstraints'][transformedConstraint]
+
+    def _find_parent_disjunct(self, constraint):
+        # traverse up until we find the disjunct this constraint lives on
+        parent_disjunct = constraint.parent_block()
+        while type(parent_disjunct) not in (_DisjunctData, SimpleDisjunct):
+            if parent_disjunct is None:
+                raise GDP_Error(
+                    "Constraint %s is not on a disjunct and so was not "
+                    "transformed" % constraint.name)
+            parent_disjunct = parent_disjunct.parent_block()
+
+        return parent_disjunct
+
+    def _get_constraint_transBlock(self, constraint):
+        parent_disjunct = self._find_parent_disjunct(constraint)
+        transBlock = parent_disjunct._transformation_block
```

Comment:
Similar to @qtothec's comment above:  is `_transformation_block` guaranteed to be present?  Instead, should this be
```python
transBlock = getattr(parent_disjunct, '_transformation_block', None)
```
(similar questions for the other information methods below)

Reply:
I think it is guaranteed here because `_find_parent_disjunct` can only return a _DisjunctData or SimpleDisjunct, either of which will have it.